sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/MLflowDataObject.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2023 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.dataobject
+
+import io.smartdatalake.config.SdlConfigObject.DataObjectId
+import io.smartdatalake.config.{ FromConfigFactory, InstanceRegistry }
+import io.smartdatalake.util.misc.SmartDataLakeLogger
+import com.typesafe.config.Config
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.util.mlflow.{ MLflowPythonSparkEntryPoint, MLflowPythonUtil }
+import io.smartdatalake.workflow.ActionPipelineContext
+import org.apache.spark.sql.{ DataFrame }
+
+// TODO: model according to experiment definition
+case class MLflowExperiment(experimentName: String)
+
+/**
+ * [[DataObject]] for interaction with MLflow
+ *
+ * @param trackingURI Uri of the MLflow server. Default is http://localhost:5000.
+ * @param experimentName The name of the experiment stored in MLflow.
+ */
+case class MLflowDataObject(
+                             override val id: DataObjectId,
+                             trackingURI: String,
+                             experimentName: String,
+                             modelName: Option[String] = None,
+                             modelDescription: Option[String] = None,
+                             override val metadata: Option[DataObjectMetadata] = None
+                           )(@transient implicit val instanceRegistry: InstanceRegistry)
+  extends DataObject
+    with CanCreateSparkDataFrame
+    with SmartDataLakeLogger {
+
+  // entry point for accessing dynamically Java objects living inside the JVM
+  var entryPoint: Option[MLflowPythonSparkEntryPoint] = None

```

Comment:
When accessing this DataObject from several Actions we might become problems with entryPoint and pythonMLFlowApi variables. It would be more logical to move this "state" to the action. We just let the MLFlowDataObject create pythonMLflowApi (or better pythonMLflowClient?) by a method get MLFlowClient, and keep the object in the the MLFlow*Action

Reply:
Good point. Renamed the variable and moved the `entryPoint` to the Action.